src/parser/keyword.c

```diff
@@ -527,9 +527,28 @@ static KEYWORD_RECORD *pt_find_keyword (const char *text);
 static int keyword_cmp (const void *k1, const void *k2);
 
 
+/* The GET_KEYWORD_HASH_VALUE() macro is the definition of the djb2 algorithm as a macro.
+ * Refer to the string_hash() function implemented in the libcubmemc.c file.
+ */
+#define GET_KEYWORD_HASH_VALUE(h,s) do { \
+      unsigned char* p = (unsigned char*)(s);   \
+      for((h) = 5381;  *p; p++ )                \
+        {                                       \
+             (h) = (((h) << 5) + (h)) + *p; /* hash * 33 + c */ \
+        }                                       \
+  } while(0)
+
 static int
 keyword_cmp (const void *k1, const void *k2)
 {
+  if (((KEYWORD_RECORD *) k1)->hash_value > ((KEYWORD_RECORD *) k2)->hash_value)
+    {
+      return 1;
+    }
+  if (((KEYWORD_RECORD *) k1)->hash_value < ((KEYWORD_RECORD *) k2)->hash_value)
+    {
+      return -1;
+    }
```

Comment:
```suggestion
    int cmp = ((KEYWORD_RECORD *) k1)->hash_value - ((KEYWORD_RECORD *) k2)->hash_value;
    
    if (cmp == 0)
      {
        cmp = strcmp (((KEYWORD_RECORD *) k1)->keyword, ((KEYWORD_RECORD *) k2)->keyword);
      }
      
    return cmp;
```

Reply:
I will change it.
Thanks for the comments.